### ISSUE TITLE
:: lost in nested names when using markdown

### ISSUE THREAD

**Issue by jwakely** [doxygen/doxygen#8755]
**Describe the bug**

A comment containing <code>\`foo\<bar\>::baz\`</code> gets rendered as `foo<bar>baz` when AUTOLINK_SUPPORT=NO is used.

The output is correct when using `@c foo<bar>::baz` instead of using markdown.

**Expected behavior**

It should be `foo<bar>::baz` whether using markdown backticks or `@c`.

**To Reproduce**
[doxygen-colons.tar.gz](https://github.com/doxygen/doxygen/files/7093271/doxygen-colons.tar.gz)

**Version**

1.9.3 (4bf942faa3128099a4fbc47162e11b0545487fd5)

**Comment by albert-github** [comment]
The complete example is a bit large but when we create, for tests, as smaller example like:
```
/// \file

/**
  * `make_unsigned<difference_type>::type`
  *
  * \c make_unsigned<difference_type>::type
  */
  using size_type = typename _Size<_Alloc, difference_type>::type;
```
and we run this with `doxygen -d markdown`, we see:
```
======== Markdown =========
---- input -------

`make_unsigned<difference_type>::type`

\c make_unsigned<difference_type>::type

---- output -----

<tt>make_unsigned\<difference_type\>::type</tt>

\c make_unsigned<difference_type>::type

=========
```
Note: the escaped `<` and `>` in the case of the back tacks / code span.
when this is run through the `doctokenizer.l` we see (only relevant parts):
```
--accepting rule at line 611 ("<tt>")
--accepting rule at line 630 ("make_unsigned")
--accepting rule at line 521 ("\<")
--accepting rule at line 630 ("difference_type")
--accepting rule at line 521 ("\>")
--accepting rule at line 630 ("::type")
--accepting rule at line 611 ("</tt>")
--accepting rule at line 669 ("
")
--accepting rule at line 520 ("\c")
--accepting rule at line 668 (" ")
--accepting rule at line 630 ("make_unsigned<difference_type>::type")
```

In other words in case of the back ticks the `::type` is seen as a new word and later on in the processing as a potential global class name etc.

I've just pushed a proposed patch, pull request #8756

**Comment by jwakely** [comment]
> The complete example is a bit large

Sorry, it came from an even bigger example, and it took me a while to figure out that the combination of markdown and AUTOLINK_SUPPORT=NO was the trigger.

> I've just pushed a proposed patch, pull request #8756

Thanks, I'll give this a try.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
